br/pkg/storage/ks3.go

```diff
@@ -0,0 +1,727 @@
+				Objects: objects,
+				Quiet:   boolP(false),
+			},
+		}
+		_ = rs.svc.DeleteObjects(input)
+		files = files[len(batch):]
+	}
+	return nil
+}
+
+func boolP(b bool) *bool {
+	return &b
+}
+
+// FileExists check if file exists on s3 storage.
+func (rs *KS3Storage) FileExists(_ context.Context, file string) (bool, error) {
+	input := &s3.HeadObjectInput{
+		Bucket: aws.String(rs.options.Bucket),
+		Key:    aws.String(rs.options.Prefix + file),
+	}
+
+	_, err := rs.svc.HeadObject(input)
+	if err != nil {
+		if aerr, ok := errors.Cause(err).(awserr.Error); ok { // nolint:errorlint
+			switch aerr.Code() {
+			case s3.ErrCodeNoSuchBucket, s3.ErrCodeNoSuchKey, notFound:
+				return false, nil
+			}
+		}
+		return false, errors.Trace(err)
+	}
+	return true, nil
+}
+
+// WalkDir traverse all the files in a dir.
+//
+// fn is the function called for each regular file visited by WalkDir.
+// The first argument is the file path that can be used in `Open`
+// function; the second argument is the size in byte of the file determined
+// by path.
+func (rs *KS3Storage) WalkDir(_ context.Context, opt *WalkOption, fn func(string, int64) error) error {
+	if opt == nil {
+		opt = &WalkOption{}
+	}
+	prefix := path.Join(rs.options.Prefix, opt.SubDir)
+	if len(prefix) > 0 && !strings.HasSuffix(prefix, "/") {
+		prefix += "/"
+	}
+
+	if len(opt.ObjPrefix) != 0 {
+		prefix += opt.ObjPrefix
+	}
+
+	maxKeys := int64(1000)
+	if opt.ListCount > 0 {
+		maxKeys = opt.ListCount
+	}
+	req := &s3.ListObjectsInput{
+		Bucket:  aws.String(rs.options.Bucket),
+		Prefix:  aws.String(prefix),
+		MaxKeys: int64p(maxKeys),
+	}
+
+	for {
+		// FIXME: We can't use ListObjectsV2, it is not universally supported.
+		// (Ceph RGW supported ListObjectsV2 since v15.1.0, released 2020 Jan 30th)
+		// (as of 2020, DigitalOcean Spaces still does not support V2 - https://developers.digitalocean.com/documentation/spaces/#list-bucket-contents)
+		res, err := rs.svc.ListObjects(req)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		for _, r := range res.Contents {
+			// https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html#AmazonS3-ListObjects-response-NextMarker -
+			//
+			// `res.NextMarker` is populated only if we specify req.Delimiter.
+			// Aliyun OSS and minio will populate NextMarker no matter what,
+			// but this documented behavior does apply to AWS S3:
+			//
+			// "If response does not include the NextMarker and it is truncated,
+			// you can use the value of the last Key in the response as the marker
+			// in the subsequent request to get the next set of object keys."
+			req.Marker = r.Key
+
+			// when walk on specify directory, the result include storage.Prefix,
+			// which can not be reuse in other API(Open/Read) directly.
+			// so we use TrimPrefix to filter Prefix for next Open/Read.
+			path := strings.TrimPrefix(*r.Key, rs.options.Prefix)
+			// trim the prefix '/' to ensure that the path returned is consistent with the local storage
```

Comment:
can you sort out these comment 😂

Reply:
sorry I don't understand your suggestion

Reply:
Because you are using another package, these comments may no longer be applicable. So maybe you can delete it directly or modify it.